src/main/java/Sprites.java

```diff
@@ -16,18 +17,49 @@
      */
     public Sprites(JSONObject stage) {
         JSONArray stageChildren = stage.optJSONArray("children");
+        spriteMap = new HashMap<String, JSONObject>();
         if (stageChildren == null) {
-            stageChildren = new JSONArray();
+            return;
         }
-        spriteMap = new HashMap<String, JSONObject>();
         for (int i = 0; i < stageChildren.length(); i++) {
             JSONObject child = stageChildren.optJSONObject(i);
             if (isSprite(child)) {
-                spriteMap.put(child.getString("objName"), child);
+                addSpriteToSpriteMap(child);
             }
         }
     }
     /**
+     * Add sprite to spriteMap. Guarantee unique name.
+     * @param sprite A Sprite JSONObject
+     */
+    private void addSpriteToSpriteMap(JSONObject sprite) {
```

Comment:
I like what you're doing here, but I think there's a shorter, cleaner way to write it. What if we do the following:
```
if the map already contains the key:
    if the key ends in a digit:
        increment the digit and try to add again (recurse?)
    else
        add a zero to the end and try to add again (recurse?)
```
If this seems more confusing (instead of less), then don't worry about it.